src/backend/projects/project_crud.py

```diff
@@ -315,29 +396,175 @@ def update_project_with_upload(
                 detail=f"{task_count} tasks were created before the following error was thrown: {e}, on feature: {feature}",
             )
 
-
 # ---------------------------
 # ---- SUPPORT FUNCTIONS ----
 # ---------------------------
 
-def create_task_grid(projectid: int, error_detail: str):
+def read_xlsforms(
+    db: Session,
+    directory: str,
+):
+    """Read the list of XLSForms from the disk"""
+    xlsforms = list()
+    for xls in os.listdir(directory):
+        if xls.endswith(".xls"):
```

Comment:
Many, many people (including me) use .xlsx files for xlsforms. I forget exactly what it was, but I encountered some limitation in the .xls spec for some ODK data once (it may have been submissions rather than xlsforms), and since then I've avoided the .xls format altogether.

Reply:
@robsavoye this one is pretty important I think. A _lot_ of people have libraries of xlsforms in .xlsx format. Go ahead and merge, but please keep in mind that this could be a really frustrating limitation for anyone modifying forms or providing their own.

Reply:
Interesting, as I've had the opposite results with xls vs xls. I can extend the code to deal with both.

Reply:
I'd appreciate that; obviously working with either/or would be preferable.